docs/README.md

```diff
@@ -1,8 +1,79 @@
-# test123
- hello testing
+---
+layout: default
+title:  "Kubernetes Credentials Provider Plugin"
+---
 
-examples
+The *Kubernetes Credentials Provider* is a [Jenkins](https://jenkins.io) plugin to enable the retreival of [Credentials](https://plugins.jenkins.io/credentials) directly from Kubernetes.
 
+The plugin supports most common credential types and defines an [`extension point`](https://jenkins.io/doc/developer/extensions/kubernetes-credentials-provider/) that can be implemented by other plugins to add support for custom Credential types. 
+
+# Using
+
+### Pre-requisites
+
+- Jenkins must be running in a kubernetes cluster
+- The pod running Jenkins must have a service account with a role that sets the following:
+  - get/watch/list permissions for `secrets`[^AWS] 
+
+[^AWS]: it is reported that running in KOPS on AWS you will also need permissions to get/watch/list `configmaps`
+
+Because granting these permissions for secrets is not something that should be done lightly it is highly advised that you run Jenkins in a unique namespace.
```

Comment:
Updated to 

Because granting these permissions for secrets is not something that should be done lightly it is highly advised for security reasons that you both create a unique service account to run Jenkins as, and run Jenkins in a unique namespace.

Reply:
NIT: no comma before an 'and' ;-)